packages/salesforcedx-vscode-core/src/commands/baseDeployRetrieve.ts

```diff
@@ -43,6 +43,18 @@ import { createComponentCount, formatException } from './util';
 type DeployRetrieveResult = DeployResult | RetrieveResult;
 type DeployRetrieveOperation = MetadataApiDeploy | MetadataApiRetrieve;
 
+async function getApiVersion(): Promise<string | undefined> {
+  const originalPath = process.cwd();
+  const workspacePath = getRootWorkspacePath();
+  process.chdir(workspacePath);
+  const aggregator = await ConfigAggregator.create();
+  const apiVersion = aggregator.getPropertyValue(
+    OrgConfigProperties.ORG_API_VERSION
+  );
+  process.chdir(originalPath);
```

Comment:
It's be nice to have a single util for getting the correctly chdir'd configAggregator

Reply:
Yep.  Created the WI for follow up.  Ive attempted a few times now, only to encounter various build issues.  Will be good to look at this together as a follow up item.